packages/fluentui/react-northstar/src/components/Chat/ChatMessage.tsx

```diff
@@ -477,24 +492,74 @@ export const ChatMessage: ComponentWithAs<'div', ChatMessageProps> &
   });
 
   const detailsElement = createShorthand(ChatMessageDetails, details, {
-    defaultProps: () => ({ mine }),
+    defaultProps: () => ({ compact, mine }),
   });
 
-  const readStatusElement = createShorthand(ChatMessageReadStatus, readStatus, {});
-
-  const headerElement = createShorthand(ChatMessageHeader, header || {}, {
-    overrideProps: () => ({
-      content: (
-        <>
-          {authorElement}
-          {timestampElement}
-          {detailsElement}
-          {reactionGroupPosition === 'start' && reactionGroupElement}
-        </>
-      ),
-    }),
+  const readStatusElement = createShorthand(ChatMessageReadStatus, readStatus, {
+    defaultProps: () => ({ compact }),
   });
 
+  let layout = <></>;
+  if (compact) {
+    const headerElement = createShorthand(ChatMessageHeader, header);
+
+    const bodyElement = createShorthand(ChatMessageCompactBody, compactBody || {}, {
+      overrideProps: () => ({
+        children: (
+          <>
+            <Flex.Item grow={1}>
+              <div>
+                {authorElement}
+                {messageContent}
+              </div>
+            </Flex.Item>
+            {timestampElement}
+            {detailsElement}
+            {badgeElement}
+          </>
+        ),
+        space: 'between',
+      }),
+    });
```

Comment:
```suggestion
    const bodyElement = Box.create(compactBody, {
      defaultProps: () =>
        getA11Props('compactBody', {
          className: chatMessageSlotClassNames.compactBody,
          styles: resolvedStyles.compactBody,
        }),
      overrideProps: () => ({
        children: (
          <>
            <Flex.Item grow={1}>
              <div>
                {authorElement}
                {messageContent}
              </div>
            </Flex.Item>
            {timestampElement}
            {detailsElement}
            {badgeElement}
          </>
        ),
      }),
    });
```

Let's do this change and:
- Add `compactBody: {},` to ChatMessage defaultProps
- Remove ChatMessageCompactBody.tsx, and change ChatMesage's compactBody prop's type to be `compactBody?: ShorthandValue<BoxProps>;`
- Add this to ChatMessageStyles.ts 
```tsx
  compactBody: (): ICSSInJSStyle => ({
    display: 'flex',
    justifyContent: `space-between`,
  }),
```

`compactBody` doesn't have its own prop, and we only need it for styling its inner content. So there's no special need to give it its own component.
And I choose Box instead of Flex because Flex does not have `create` function

Reply:
Agree, the main reason to hoist out to separate components is for stlye caching but that would imply a separate styles file too

Reply:
As discussed with @YuanboXue-Amber  we might want to discuss about styles caching internally, but let's keep it simple for now

Reply:
- Added `compactBody: {}` to `defaultProps`
- Added `compactBody` to `propTypes`
- Remove `ChatMessageCompactBody`-component
- Added `compactBody` slot styles

Some questions:
- Why would this one need `getA11Props` when none of the other slots (`header`, `content`, ...) use it?
- Is there a difference/preference between using `content` and `children` for `Box`?

Reply:
thanks for the changes!

- Is there a difference/preference between using content and children for Box
Good catch! I think we should use `content` instead of `children`. Could you make this change? Sorry I made this mistake.
Reason: `children` can override `content` so it's more customisable

- Why would this one need getA11Props when none of the other slots (header, content, ...) use it?
`getA11Props` does this: ChatMessage component can have custom a11y behavior defined through `accessibility` prop. And it is possible to add a11y behavior to each slot in the custom defined `accessibility`. In this case `getA11Props` is needed to pass in the a11y for each slot. 
So it is needed for all slot. The other slot should use it, so it's in correct that they did not currently. But it doesn't have big impact because TMP doesn't use custom defined `accessibility` prop.

Reply:
Done, thanks for the details.

- Changed to `content`
- Added `getA11Props`